CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/apis/OpenAiTests.cs

```diff
@@ -162,6 +164,49 @@ public class OpenAiTests {
             return emotionalChatResponse;
         }
 
+
+        [Fact]
+        public async Task ExampleUsage5_ImageToText() {
+            var openAi = new OpenAi(await IoC.inject.GetAppSecrets().GetSecret("OpenAiKey"));
+
+            var prompt = "A picture of a dog";
+            var result = await openAi.TextToImage(new OpenAi.Image.Request() { prompt = prompt });
+            var url = result.data.First().url;
+            Assert.NotEmpty(url);
+
+            var messages = new List<VisionGpt.Line>() {
+                new VisionGpt.Line(VisionGpt.Role.system, content: "You are a helpful assistant designed to output JSON.")
+            };
+
+            var yesNoResponseFormat = new YesNoResponse() {
+                confidence = 100,
+                inputQuestionInterpreted = "Is there a cat in the image?",
+                yesNoAnswer = true,
+                explanation = "The cat is in the picture because I see a small feline with whiskers."
+            };
+            messages.AddImageURL(url);
+            messages.AddUserLineWithJsonResultStructure("Is there a dog in the picture?", yesNoResponseFormat);
+
+            // Send the messages to the AI and get the response:
+            var response = await openAi.ImageToText(new VisionGpt.Request(messages));
+            VisionGpt.Line newLine = response.choices.Single().message;
+            messages.Add(newLine);
+
+            // Parse newLine.content as a YesNoResponse:
+            var yesNoResponse = newLine.ParseNewLineContentAsJson<YesNoResponse>();
+
+            // Dogs can look up, lets hope the AI knows that too:
+            Assert.True(yesNoResponse.yesNoAnswer);
+            // Since the input question is very short the interpretation will be the same string:
+            Assert.Equal("Is there a dog in the picture?", yesNoResponse.inputQuestionInterpreted);
+            // The AI is very confident in its answer:
+            Assert.True(yesNoResponse.confidence > 50);
```

Comment:
Since the YesNoResponse value for confidence was an integer before we added our functionality and this interface is also used in `ExampleUsage4_ChatGptJsonResponses` we are not sure whether we should change it in our new method and in the old one as well or maybe leave it as is and consider refactoring this in a future PR.

Reply:
Yes ok fine with me